tests/smoke/smoke_test.go

```diff
@@ -113,6 +114,33 @@ var _ = Describe("Smoke Tests", func() {
 				cfLogs := cf.Cf("logs", appName, "--recent")
 				return string(cfLogs.Wait().Out.Contents())
 			}, 2*time.Minute, 2*time.Second).Should(ContainSubstring("Console output from test-node-app"))
+		}
+
+		When("app source is < 64k", func() {
+			It("creates a routable app pod in Kubernetes from a source-based app", func() {
+				expectSuccessfulPush("assets/test-node-app")
+			})
+		})
+
+		When("app source is > 64k", func() {
+			var tempdir string
+			BeforeSuite(func() {
+				var err error
+				tempdir, err = ioutil.TempDir("", "cf-for-k8s-smoke-test-big-app")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(copy.Copy("assets/test-node-app", tempdir)).To(Succeed())
+				bigFileName := path.Join(tempdir, "a-big-file.txt")
+				bigFileContent := []byte(strings.Repeat("na", 32*1024) + " batman")
+				Expect(ioutil.WriteFile(bigFileName, bigFileContent, 0o600)).To(Succeed())
+			})
+
+			It("creates a routable app pod in Kubernetes from a source-based app", func() {
+				expectSuccessfulPush(tempdir)
+			})
+
+			AfterSuite(func() {
```

Comment:
Minor nit: I think this could/should be an `AfterEach` instead of an `AfterSuite`.

Reply:
I agree with Dave's comments: I looked at 3 different CF test directories, and all of them use `BeforeSuite` and `AfterSuite` in top-level `suite_test` files only. Everywhere else, they use `BeforeEach` and `AfterEach`.  If you want the `Before...` actions to apply to only the one test, you can wrap it in a Context block. Typically, we avoid wrapping a single test in a Context block, and add them when more tests are added.